api/apps/v1alpha1/nimservice_types.go

```diff
@@ -480,6 +536,37 @@ func (n *NIMService) GetDeploymentKind() string {
 	return "Deployment"
 }
 
+// GetInitContainers returns the init containers for the NIMService deployment
+func (n *NIMService) GetInitContainers() []corev1.Container {
+	if n.Spec.Proxy != nil {
+		return []corev1.Container{
```

Comment:
Wondering if we should move the generation of initContainer spec for proxy into a k8sUtil package. That way we can leverage same for NIMCache, and other NeMo services that are pulling models from NGC/HF/S3 etc.

Reply:
Cannot move the method because image of the container is based on the specific object like NIMCache, NIMService.